### DISCUSSION TITLE
Creating a new product and default variant while importing from CSV

### DISCUSSION THREAD

**Discussion by johnchiverton** [saleor/saleor#9422] [Q&A] [answered]
Where I am:
Importing data successfully from a CSV file, creating multiple new products as we go.
Manually creating a single variant using productVariantCreate and using a hard-coded product id.

Where I want to be:
For each record I import from the CSV file, create a new product AND a single default variant for that product.

I'm sure it's going to be a simple solution, but a cloud has descended on my brain and is obscuring my vision.

_Any_ help is appreciated!

**Comment by krzysztofwolski** [comment]
Hello!
At the moment there's no single mutation to do both product and variant in a single run. For sure we'll be implementing bulk importing operations, but it's not in the near future so separate mutations have to be used.

**Reply by johnchiverton** [reply]
Thanks Krzysztofwolski. Now working with separate mutations. Thanks for the pointers.

**Reply by sruthi-pooliry** [reply]
Hi @johnchiverton  how did you import data from a csv file , could you please share the steps you followed for achieving that

**Comment by timuric** [comment, marked as answer]
Please take a look at this mutation https://docs.saleor.io/docs/3.x/api-reference/products/mutations/product-bulk-create
